### PR TITLE
feat(ct): port interface checks to go

### PR DESCRIPTION
**Pull request by smartcontracts**
Ports the interface checks command to go. Updates build command to execute forge fmt and the interface checks.

### CODE REVIEW

File: packages/contracts-bedrock/scripts/checks/interfaces/main.go [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+	var hasErr int32
+	var outMtx sync.Mutex
+	fail := func(msg string, args ...any) {
+		outMtx.Lock()
+		writeStderr("❌  "+msg, args...)
+		outMtx.Unlock()
+		atomic.StoreInt32(&hasErr, 1)
+	}
+
+	sem := make(chan struct{}, runtime.NumCPU())
```
Comment (tynes) [inline packages/contracts-bedrock/scripts/checks/interfaces/main.go:95]: I recommend the following package for things like this in the future: https://pkg.go.dev/golang.org/x/sync@v0.8.0/errgroup

If you look at the example code, it makes what you want to do really easy, you don't need to manage a semaphore or atomics

Comment (smartcontracts) [reply packages/contracts-bedrock/scripts/checks/interfaces/main.go:95]: Gotcha, I just copied another script but happy to switch to this

File: packages/contracts-bedrock/scripts/checks/interfaces/main_test.go [hunk 2]
```diff
@@ -0,0 +1,295 @@
+package main
```
Comment (Inphi) [inline packages/contracts-bedrock/scripts/checks/interfaces/main_test.go:1]: Is this tested somewhere in ci?

Comment (smartcontracts) [reply packages/contracts-bedrock/scripts/checks/interfaces/main_test.go:1]: Ah good point, I don't think that it is. Mind if I add this in a separate PR that includes testing in CI for all of the other Go checks?

Comment (Inphi) [reply packages/contracts-bedrock/scripts/checks/interfaces/main_test.go:1]: That's OK.

**Review by Inphi — APPROVED** [review]

**Pull request merged**